trieste/models/gpflow/sampler.py

```diff
@@ -874,19 +881,26 @@ def __call__(self, x: TensorType) -> TensorType:  # [N, B, d] -> [N, B, 1]
             """,
         )
 
-        flat_x, unflatten = flatten_leading_dims(x)  # [N*B, d]
-        flattened_feature_evaluations = self._feature_functions(flat_x)  # [N*B, m]
-        feature_evaluations = unflatten(flattened_feature_evaluations)  # [N, B, m]
+        flat_x, unflatten = flatten_leading_dims(x)  # [N*B, D]
+        flattened_feature_evaluations = self._feature_functions(
+            flat_x
+        )  # [N*B, F + M] or [L, N*B, F + M]
+        # ensure tensor is always rank 3
+        rank3_shape = tf.concat([[1], tf.shape(flattened_feature_evaluations)], axis=0)[-3:]
```

Comment:
we have a flatten / unflatten helper function in utils, perhaps could be used here? (not sure)

Reply:
Flatten/unflatten is already used for `x`. I tried to also use it here, but can't see a way where it helps with reshaping/rank. There may be a way if I also change the subsequent lines, e.g. transpose/permutation, but that doesn't seem like a big win.